IllegalStateException crash in autofill: onFillRequest$5.accept

Google Pixel, Android 9

```
io.reactivex.exceptions.OnErrorNotImplementedException: 
 
  at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept (Functions.java:704)
 
  at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept (Functions.java:701)
 
  at io.reactivex.internal.observers.LambdaObserver.onError (LambdaObserver.java:77)
 
  at io.reactivex.internal.observers.LambdaObserver.onNext (LambdaObserver.java:67)
 
  at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext (ObservableMap.java:62)
 
  at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext (ObservableFilter.java:52)
 
  at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext (ObservableMap.java:62)
 
  at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext (ObservableMap.java:62)
 
  at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext (ObservableFilter.java:52)
 
  at io.reactivex.subjects.PublishSubject$PublishDisposable.onNext (PublishSubject.java:308)
 
  at io.reactivex.subjects.PublishSubject.onNext (PublishSubject.java:228)
 
  at mozilla.lockbox.flux.Dispatcher.dispatch (Dispatcher.kt:22)
 
  at mozilla.lockbox.presenter.AutofillFilterPresenter$onViewReady$2.invoke (AutofillFilterPresenter.kt:42)
 
  at mozilla.lockbox.presenter.AutofillFilterPresenter$onViewReady$2.invoke (AutofillFilterPresenter.kt:34)
 
  at mozilla.lockbox.presenter.AutofillFilterPresenter$sam$io_reactivex_functions_Consumer$0.accept (Unknown Source:2)
 
  at io.reactivex.internal.observers.LambdaObserver.onNext (LambdaObserver.java:63)
 
  at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext (ObservableMap.java:62)
 
  at io.reactivex.subjects.PublishSubject$PublishDisposable.onNext (PublishSubject.java:308)
 
  at io.reactivex.subjects.PublishSubject.onNext (PublishSubject.java:228)
 
  at mozilla.lockbox.view.AutofillFilterFragment.onDestroyView (AutofillFilterFragment.kt:88)
 
  at androidx.fragment.app.Fragment.performDestroyView (Fragment.java:2779)
 
  at androidx.fragment.app.FragmentManagerImpl.moveToState (FragmentManagerImpl.java:948)
 
  at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState (FragmentManagerImpl.java:1227)
 
  at androidx.fragment.app.FragmentManagerImpl.moveToState (FragmentManagerImpl.java:1293)
 
  at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange (FragmentManagerImpl.java:2612)
 
  at androidx.fragment.app.FragmentManagerImpl.dispatchDestroy (FragmentManagerImpl.java:2603)
 
  at androidx.fragment.app.FragmentController.dispatchDestroy (FragmentController.java:329)
 
  at androidx.fragment.app.FragmentActivity.onDestroy (FragmentActivity.java:353)
 
  at androidx.appcompat.app.AppCompatActivity.onDestroy (AppCompatActivity.java:208)
 
  at mozilla.lockbox.view.AutofillRootActivity.onDestroy (AutofillRootActivity.kt:28)
 
  at android.app.Activity.performDestroy (Activity.java:7403)
 
  at android.app.Instrumentation.callActivityOnDestroy (Instrumentation.java:1306)
 
  at android.app.ActivityThread.performDestroyActivity (ActivityThread.java:4481)
 
  at android.app.ActivityThread.handleDestroyActivity (ActivityThread.java:4515)
 
  at android.app.servertransaction.DestroyActivityItem.execute (DestroyActivityItem.java:39)
 
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:145)
 
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1816)
 
  at android.os.Handler.dispatchMessage (Handler.java:106)
 
  at android.os.Looper.loop (Looper.java:193)
 
  at android.app.ActivityThread.main (ActivityThread.java:6718)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
Caused by: java.lang.IllegalStateException: 
 
  at android.service.autofill.FillCallback.assertNotCalled (FillCallback.java:105)
 
  at android.service.autofill.FillCallback.onSuccess (FillCallback.java:58)
 
  at mozilla.lockbox.LockboxAutofillService$onFillRequest$5.accept (LockboxAutofillService.kt:121)
 
  at mozilla.lockbox.LockboxAutofillService$onFillRequest$5.accept (LockboxAutofillService.kt:38)
 
  at io.reactivex.internal.observers.LambdaObserver.onNext (LambdaObserver.java:63)
```

https://play.google.com/apps/publish/?account=7083182635971239206#AndroidMetricsErrorsPlace:p=mozilla.lockbox&appid=4972100280256015711&appVersion=PRODUCTION&clusterName=apps/mozilla.lockbox/clusters/7668289a&detailsAppVersion=PRODUCTION&detailsSpan=7

--- a/app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt
+++ b/app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt
@@ -102,6 +102,7 @@ class LockboxAutofillService(
             .addTo(compositeDisposable)
 
         autofillStore.autofillActions
+            .take(1)
             .map {
                 when (it) {
                     is AutofillAction.Complete -> builder.buildFilteredFillResponse(this, listOf(it.login)).asOptional()
@@ -117,9 +118,11 @@ class LockboxAutofillService(
                 }.asOptional()
             }
             .filterNotNull()
-            .subscribe {
+            .subscribe( {
                 callback.onSuccess(it.value)
-            }
+            }, {
+                log.error(throwable = it)
+            })
             .addTo(compositeDisposable)
     }
 

--- a/app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt
+++ b/app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt
@@ -118,7 +118,7 @@ class LockboxAutofillService(
                 }.asOptional()
             }
             .filterNotNull()
-            .subscribe( {
+            .subscribe({
                 callback.onSuccess(it.value)
             }, {
                 log.error(throwable = it)

